src/net/network_service_test.cc

```diff
@@ -0,0 +1,53 @@
+#include <net/network_service_impl.h>
+
+#include <base/aliases.h>
+#include <base/thread.h>
+#include <net/end_point.h>
+#include <net/socket.h>
+#include <net/passive.h>
+
+#include <third_party/gtest/exported/include/gtest/gtest.h>
+
+namespace dist_clang {
+namespace net {
+
+class NetworkServiceTest : public ::testing::Test {
+ public:
+  virtual void SetUp() override {
+    NetworkService::SetFactory<NetworkService::DefaultFactory>();
+  }
+};
+
+TEST_F(NetworkServiceTest, ConnectTimedOut) {
+  auto service = net::NetworkService::Create(0, 0, 1, 1);
+  EndPointPtr unreachable = EndPoint::TcpHost("example.com", 12345, false);
```

Comment:
This is pretty silly, I know :) But actually, it seems to do the basic check on a machine with IPv4 connectivity (see https://travis-ci.org/andrey-malets/dist-clang/builds/143170185).

I see three possible solutions probably better than this:
1. Mock entire network stack with a proper library. Requires such a library :)
2. Mangle with `iptables` blocking connections to localhost in the test. Requires super-user privileges to run the test, which I do not find best, plus some OS-specific glue.
3. Disable the test :)